### PR TITLE
[AIRFLOW-3773] Fix /refresh_all endpoint

### PR DESCRIPTION
**Pull request by astahlman**
Make sure you have checked _all_ steps below.

### Jira

- [ X ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3773

### Description

- [ X ] Here are some details about my PR, including screenshots of any UI changes:

Call `sync_perm_for_dag` for each DAG in the DagBag (`dag_id` is a
required argument).

### Tests

- [ X ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

I looked for a test suite for the web UI, but it seems the existing
tests have all been disabled since the switch to FAB. I've created a new
class for FAB tests and added a test to exercise the `/refresh_all`
endpoint.

### Commits

- [ X ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ X ] Passes `flake8`

### CODE REVIEW

**Comment by astahlman** [comment]
cc @feng-tao

File: tests/core.py [hunk 1]
```diff
@@ -1867,6 +1867,41 @@ def tearDown(self):
         self.dag_bash.clear(start_date=DEFAULT_DATE, end_date=timezone.utcnow())
 
 
+class FabTests(unittest.TestCase):
```
Comment (feng-tao) [inline tests/core.py]: could you move the test to test_views.py(https://github.com/apache/airflow/blob/master/tests/www/test_views.py#L517-L519) ? We have put most of the FAB tests to that place.

**Comment by feng-tao** [comment]
Minor suggestion for the test location. Otherwise LGTM

File: tests/core.py (hunk 1 above)
Comment (feng-tao) [reply tests/core.py]: We have already had the setup code in that test. So you only need the test_refresh_all function.

File: airflow/www/views.py [hunk 2]
```diff
@@ -1687,8 +1687,9 @@ def refresh(self, session=None):
     @action_logging
     def refresh_all(self):
         dagbag.collect_dags(only_if_updated=False)
-        # sync permissions for all dags
-        appbuilder.sm.sync_perm_for_dag()
+        for dag_id in dagbag.dags:
+            # sync permissions for all dags
+            appbuilder.sm.sync_perm_for_dag(dag_id)
```
Comment (feng-tao) [inline airflow/www/views.py:1692]: @astahlman , since you are at this function, maybe we should consider add a check and see whether the user has permission to do refresh_all view.

The AirflowSecurityManager provides a way to check whether the user's role(https://github.com/apache/airflow/blob/master/airflow/www/security.py#L283). But we could do it in a different pr.

**Comment by feng-tao** [comment]
lgtm
